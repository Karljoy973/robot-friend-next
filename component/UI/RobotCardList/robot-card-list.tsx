import React from "react";
import RobotCard, { RobotCardArgs } from "../RobotCard/robot-card";

type RobotCardListArgs = { data: RobotCardArgs[] };
const RobotCardList = ({ data }: RobotCardListArgs) => {
	let items = data.map((e) => (
		<li className="mx-4 my-2" key={e.name + e.email + Date.now()}>
			<RobotCard name={e.name} email={e.email} />
		</li>
	));
	return (
		<div>
			<ul className="flex flex-wrap">{items}</ul>
		</div>
	);
};
 
export default RobotCardList;