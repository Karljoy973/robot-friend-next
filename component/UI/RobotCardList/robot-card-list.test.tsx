import React from "react";
import RobotCardList from "./robot-card-list";
import { render, screen } from "@testing-library/react";

let db = [
	{
		email: "toto@toto.com",
		name: "toto",
		robotId: `${Date.now()}`,
	},
	{
		email: "tata@tata.com",
		name: "tata",
		robotId: `${Date.now()}`,
	},
	{
		email: "titi@titi.com",
		name: "titi",
		robotId: `${Date.now()}`,
	},
	{
		email: "tutu@tutu.com",
		name: "tutu",
		robotId: `${Date.now()}`,
	},
];
beforeEach(() => {
	render(<RobotCardList data={db} />);
});

it("shall display a list of Robot Cards", async () => {
	// je dois avoir n RobotCards dans le body ou dans un de ses enfants
	let list = await screen.findByRole("list");
	expect(list.childElementCount).toEqual(db.length);
});

it("shall display a grid of robot cards correctly", () => {
	expect(screen.getByRole("list", {})).toMatchSnapshot();
});