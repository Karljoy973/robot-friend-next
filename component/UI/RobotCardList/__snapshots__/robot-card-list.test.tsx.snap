// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shall display a grid of robot cards correctly 1`] = `
<ul
  class="flex flex-wrap"
>
  <li
    class="mx-4 my-2"
  >
    <div
      class="flex w-3xs flex-col justify-center items-center rounded-2xl bg-teal-200"
      id="0195643e-ab21-74c7-b503-f27371c2476a"
    >
      <img
        class="my-4"
        src="https://robohash.org/0195643e-ab21-74c7-b503-f27371c2476a?size=200x200"
      />
      <span
        class="grow flex flex-col justify-center items-center my-4"
      >
        <h2
          class="w-fit text-xl font-semibold font-[system-ui] text-black"
        >
          toto
        </h2>
        <p
          class="font-[system-ui] text-black font-medium"
        >
          toto@toto.com
        </p>
      </span>
    </div>
  </li>
  <li
    class="mx-4 my-2"
  >
    <div
      class="flex w-3xs flex-col justify-center items-center rounded-2xl bg-teal-200"
      id="0195643e-ab25-70f9-8b29-066938b392fa"
    >
      <img
        class="my-4"
        src="https://robohash.org/0195643e-ab25-70f9-8b29-066938b392fa?size=200x200"
      />
      <span
        class="grow flex flex-col justify-center items-center my-4"
      >
        <h2
          class="w-fit text-xl font-semibold font-[system-ui] text-black"
        >
          tata
        </h2>
        <p
          class="font-[system-ui] text-black font-medium"
        >
          tata@tata.com
        </p>
      </span>
    </div>
  </li>
  <li
    class="mx-4 my-2"
  >
    <div
      class="flex w-3xs flex-col justify-center items-center rounded-2xl bg-teal-200"
      id="0195643e-ab27-755b-9a99-3f9428be30d0"
    >
      <img
        class="my-4"
        src="https://robohash.org/0195643e-ab27-755b-9a99-3f9428be30d0?size=200x200"
      />
      <span
        class="grow flex flex-col justify-center items-center my-4"
      >
        <h2
          class="w-fit text-xl font-semibold font-[system-ui] text-black"
        >
          titi
        </h2>
        <p
          class="font-[system-ui] text-black font-medium"
        >
          titi@titi.com
        </p>
      </span>
    </div>
  </li>
  <li
    class="mx-4 my-2"
  >
    <div
      class="flex w-3xs flex-col justify-center items-center rounded-2xl bg-teal-200"
      id="0195643e-ab28-74a9-b7fd-56cb9c446063"
    >
      <img
        class="my-4"
        src="https://robohash.org/0195643e-ab28-74a9-b7fd-56cb9c446063?size=200x200"
      />
      <span
        class="grow flex flex-col justify-center items-center my-4"
      >
        <h2
          class="w-fit text-xl font-semibold font-[system-ui] text-black"
        >
          tutu
        </h2>
        <p
          class="font-[system-ui] text-black font-medium"
        >
          tutu@tutu.com
        </p>
      </span>
    </div>
  </li>
</ul>
`;
